src/notifier.rs

```diff
@@ -0,0 +1,55 @@
+use crate::constants::{MAX_NUMBER_OF_TRIES, SECS_TO_SLEEP_AFTER_REQUEST_ERROR};
+use serde_json::json;
+use std::error::Error;
+use std::thread::sleep;
+use std::time::Duration;
+
+pub struct TelegramNotifier {
+    token: String,
+    chat_id: String,
+}
+
+impl TelegramNotifier {
```

Comment:
Just `Telegram` would be enough. `Notifier::TelegramNotifier` is a tautology. Also minor comment.

Reply:
In Rust by convention we usually access struct without any prefixes, so in the place of usage you will just see Telegram. Its purpose may not be obvious at first glance. That's why I decided to call it TelegramNotifier.